tests/network/bindingplugin.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libkvconfig"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+const (
+	passtBindingName  = "passt"
+	passtSidecarImage = "registry:5000/kubevirt/network-passt-binding:devel"
+	passtNetAttDef    = "netbindingpasst"
+	passtType         = "kubevirt-passt-binding"
+)
+
+var _ = SIGDescribe("[Serial]network binding plugin", Serial, decorators.NetCustomBindingPlugins, func() {
+
+	libkvconfig.WithFeatureGate(virtconfig.NetworkBindingPlugingsGate)
+
+	Context("passt", func() {
+		libkvconfig.WithNetBindingPlugin(passtBindingName, v1.InterfaceBindingPlugin{
+			SidecarImage:                passtSidecarImage,
+			NetworkAttachmentDefinition: passtNetAttDef,
+		})
+
+		BeforeEach(func() {
+			namespace := testsuite.GetTestNamespace(nil)
+			Expect(createBasicNetworkAttachmentDefinition(namespace, passtNetAttDef, passtType)).To(Succeed())
+		})
+
+		It("can be used by a VM as its primary network", func() {
```

Comment:
To verify the CNI is working ping and low ports have to be tested.
I would update the existing passt tests to use the plugin (we can keep for now one test using the old API) instead of adding a separate bindingplugin test file.

Reply:
This test is not interested to check passt functionality in particular, it is using it as a given available example. Therefore, the checks focus on the net binding plugin aspects and less on the particular plugin functionality.

> I would update the existing passt tests to use the plugin (we can keep for now one test using the old API) instead of adding a separate bindingplugin test file.

I think we should do it in a dedicated PR when we in practice deprecate the existing passt API. But I am not that opinionated if we move the existing tests to the binding plugin, just lets do it as a follow up.

Reply:
Bottom line, the CNI introduced in this PR is not tests by e2e.

Reply:
It is tested that it was executed successfully, not that it does the correct thing.

It has unit tests that checks the code itself.

If you insist, I can move the existing passt tests, but it has support implications IMO. Only if we get confirmation to deprecate from v1.2 the current API it will be correct.
That is why I think we should handle it in a different PR.